mesonbuild/backend/ninjabackend.py

```diff
@@ -501,7 +501,7 @@ def generate_custom_target(self, target, outfile):
         deps = self.unwrap_dep_list(target)
         deps += self.get_custom_target_depend_files(target)
         desc = 'Generating {0} with a {1} command.'
-        if target.build_always:
+        if target.build_always or target.build_always_stale:
```

Comment:
Same here, replace `target.build_always` with `target.build_always_stale` and have the kwarg set `target.build_always_stale`.